appserver/payara-appserver-modules/healthcheck-metrics/src/main/java/fish/payara/healthcheck/microprofile/metrics/MicroProfileMetricsCheck.java

```diff
@@ -0,0 +1,251 @@
+import fish.payara.nucleus.healthcheck.preliminary.BaseHealthCheck;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import org.eclipse.microprofile.metrics.ConcurrentGauge;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Gauge;
+import org.eclipse.microprofile.metrics.Histogram;
+import org.eclipse.microprofile.metrics.Meter;
+import org.eclipse.microprofile.metrics.Metric;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.SimpleTimer;
+import org.eclipse.microprofile.metrics.Timer;
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.glassfish.internal.api.Globals;
+import org.jvnet.hk2.annotations.Service;
+
+@Service(name = "healthcheck-mpmetrics")
+@RunLevel(StartupRunLevel.VAL)
+public class MicroProfileMetricsCheck
+        extends BaseHealthCheck<HealthCheckMicroProfileMetricstExecutionOptions, MicroProfileMetricsChecker> {
+
+    @PostConstruct
+    public void postConstruct() {
+        postConstruct(this, MicroProfileMetricsChecker.class);
+    }
+
+    @Override
+    public HealthCheckMicroProfileMetricstExecutionOptions constructOptions(MicroProfileMetricsChecker checker) {
+        return new HealthCheckMicroProfileMetricstExecutionOptions(Boolean.valueOf(
+                checker.getEnabled()),
+                Long.parseLong(checker.getTime()),
+                asTimeUnit(checker.getUnit()),
+                checker.getMetricsScope(),
+                checker.getMetricApplicationName(),
+                checker.getMetricName());
+    }
+
+    @Override
+    protected String getDescription() {
+        return "healthcheck.description.MPmetrics";
+    }
+
+    @Override
+    protected HealthCheckResult doCheckInternal() {
+        String registryName = options.getMetricsScope();
+        List<String> metricNames = new ArrayList<>(Arrays.asList(options.getMetricName().split(",")));
+        metricNames.removeAll(Arrays.asList("", null));
+        MetricsService metricsService = Globals.getDefaultBaseServiceLocator().getService(MetricsService.class);
+        if (registryName.equalsIgnoreCase("application")) {
+            registryName = options.getMetricsApplicationName();
+        }
+
+        HealthCheckResult result = new HealthCheckResult();
+        List<String> metrics = collectMetrics(metricsService.getRegistry(registryName), metricNames);
+
+        result.add(new HealthCheckResultEntry(metrics.isEmpty() ? WARNING : GOOD,
+                metrics.isEmpty() ? "The metric you entered doesn't exist under " + registryName : metrics.stream().map(Object::toString).collect(Collectors.joining())));
```

Comment:
Can we have a few more variables? I don't find the streaming particularly intuitive here, and the result contains excess whitespace